# Toronto Neighborhoods Project

In this project, I will be exploring and clustering neighborhoods in Toronto. 

This project highlights my skills in:
- BeautifulSoup web scraping
- Dataframe manipulation with Pandas
- Mapping with Folium
- Collecting data from the Foursquare API


## Part 1: Preparing the Dataframe

First, I will scrape data from the following website to gather postal codes of various neighborhoods. 
- https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M


In [8]:
# Imports

import pandas as pd
import requests
from bs4 import BeautifulSoup as bs
from IPython.display import display, HTML # Allows for aesthetic data frame displays

print("Imports complete.")

Imports complete.


In [9]:
# Scrape data from the Wikipedia page

page = requests.get("https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M")
soup = bs(page.text, 'lxml')
table = soup.find("table", class_ = "wikitable")

In [10]:
# Transforming data into a dataframe

A = []
B = []
C = []

# In HTML, <tr> denotes a row in a table and <td> denotes each element
# These tags are used to isolate the data from the table

for row in table.findAll("tr"):
    cells = row.findAll("td")
    if len(cells) == 3: # Isolates rows from the body
        A.append(cells[0].find(text=True))
        B.append(cells[1].find(text=True))
        C.append(cells[2].find(text=True))
        
df = pd.DataFrame(A, columns = ['Postal Code'])
df['Borough'] = B
df['Neighborhood'] = C



print(df.shape)
display(df.head(10))


(288, 3)


Postal Code           Borough      Neighborhood
0         M1A      Not assigned     Not assigned

1         M2A      Not assigned     Not assigned

2         M3A        North York         Parkwoods
3         M4A        North York  Victoria Village
4         M5A  Downtown Toronto      Harbourfront
5         M5A  Downtown Toronto       Regent Park
6         M6A        North York  Lawrence Heights
7         M6A        North York    Lawrence Manor
8         M7A      Queen's Park     Not assigned

9         M8A      Not assigned     Not assigned

Now, I will remove the rows where a borough is not assigned. I will also assign the borough name to the neighborhood if one is not assigned.

Additionally, more than one neighborhood can exist in one postal code area. These rows will be combined to show a list in the neighborhood column. 

In [11]:
# Remove rows where a borough is not assigned

df.drop(df[df.Borough == "Not assigned"].index, inplace = True)
print(df.shape)
display(df.head(10))

(211, 3)


Postal Code           Borough      Neighborhood
2          M3A        North York         Parkwoods
3          M4A        North York  Victoria Village
4          M5A  Downtown Toronto      Harbourfront
5          M5A  Downtown Toronto       Regent Park
6          M6A        North York  Lawrence Heights
7          M6A        North York    Lawrence Manor
8          M7A      Queen's Park     Not assigned

10         M9A         Etobicoke  Islington Avenue
11         M1B       Scarborough             Rouge
12         M1B       Scarborough           Malvern

In [12]:
# Remove the \n that appears in some rows

df.replace('\n','', regex = True, inplace = True)


# Combine neighborhoods that have the same postcode

df2 = df.groupby(['Postal Code', 'Borough'])['Neighborhood'].apply(", ".join).reset_index()

print(df2.shape)
display(df2.head(10))

(103, 3)


Postal Code      Borough                                     Neighborhood
0         M1B  Scarborough                                   Rouge, Malvern
1         M1C  Scarborough           Highland Creek, Rouge Hill, Port Union
2         M1E  Scarborough                Guildwood, Morningside, West Hill
3         M1G  Scarborough                                           Woburn
4         M1H  Scarborough                                        Cedarbrae
5         M1J  Scarborough                              Scarborough Village
6         M1K  Scarborough      East Birchmount Park, Ionview, Kennedy Park
7         M1L  Scarborough                  Clairlea, Golden Mile, Oakridge
8         M1M  Scarborough  Cliffcrest, Cliffside, Scarborough Village West
9         M1N  Scarborough                      Birch Cliff, Cliffside West

In [13]:
# Assign the borough name to the neighborhood for any rows with no assigned neighborhood

df2.loc[df2.Neighborhood == "Not assigned", ['Neighborhood']] = df2['Borough']
print(df2.shape)
display(df2.head(10))

(103, 3)


Postal Code      Borough                                     Neighborhood
0         M1B  Scarborough                                   Rouge, Malvern
1         M1C  Scarborough           Highland Creek, Rouge Hill, Port Union
2         M1E  Scarborough                Guildwood, Morningside, West Hill
3         M1G  Scarborough                                           Woburn
4         M1H  Scarborough                                        Cedarbrae
5         M1J  Scarborough                              Scarborough Village
6         M1K  Scarborough      East Birchmount Park, Ionview, Kennedy Park
7         M1L  Scarborough                  Clairlea, Golden Mile, Oakridge
8         M1M  Scarborough  Cliffcrest, Cliffside, Scarborough Village West
9         M1N  Scarborough                      Birch Cliff, Cliffside West

The data frame is now complete and ready for future analysis.

**Here is the final data frame and shape:**

In [14]:
print(df2.shape)
display(df2)

(103, 3)


Postal Code           Borough  \
0           M1B       Scarborough   
1           M1C       Scarborough   
2           M1E       Scarborough   
3           M1G       Scarborough   
4           M1H       Scarborough   
5           M1J       Scarborough   
6           M1K       Scarborough   
7           M1L       Scarborough   
8           M1M       Scarborough   
9           M1N       Scarborough   
10          M1P       Scarborough   
11          M1R       Scarborough   
12          M1S       Scarborough   
13          M1T       Scarborough   
14          M1V       Scarborough   
15          M1W       Scarborough   
16          M1X       Scarborough   
17          M2H        North York   
18          M2J        North York   
19          M2K        North York   
20          M2L        North York   
21          M2M        North York   
22          M2N        North York   
23          M2P        North York   
24          M2R        North York   
25          M3A        North York   
26          M3B        North York   
27          M3C        North York   
28          M3H        North York   
29          M3J        North York   
..          ...               ...   
73          M6C              York   
74          M6E              York   
75          M6G  Downtown Toronto   
76          M6H      West Toronto   
77          M6J      West Toronto   
78          M6K      West Toronto   
79          M6L        North York   
80          M6M              York   
81          M6N              York   
82          M6P      West Toronto   
83          M6R      West Toronto   
84          M6S      West Toronto   
85          M7A      Queen's Park   
86          M7R       Mississauga   
87          M7Y      East Toronto   
88          M8V         Etobicoke   
89          M8W         Etobicoke   
90          M8X         Etobicoke   
91          M8Y         Etobicoke   
92          M8Z         Etobicoke   
93          M9A         Etobicoke   
94          M9B         Etobicoke   
95          M9C         Etobicoke   
96          M9L        North York   
97          M9M        North York   
98          M9N              York   
99          M9P         Etobicoke   
100         M9R         Etobicoke   
101         M9V         Etobicoke   
102         M9W         Etobicoke   

                                          Neighborhood  
0                                       Rouge, Malvern  
1               Highland Creek, Rouge Hill, Port Union  
2                    Guildwood, Morningside, West Hill  
3                                               Woburn  
4                                            Cedarbrae  
5                                  Scarborough Village  
6          East Birchmount Park, Ionview, Kennedy Park  
7                      Clairlea, Golden Mile, Oakridge  
8      Cliffcrest, Cliffside, Scarborough Village West  
9                          Birch Cliff, Cliffside West  
10   Dorset Park, Scarborough Town Centre, Wexford ...  
11                                   Maryvale, Wexford  
12                                           Agincourt  
13             Clarks Corners, Sullivan, Tam O'Shanter  
14   Agincourt North, L'Amoreaux East, Milliken, St...  
15                                     L'Amoreaux West  
16                                         Upper Rouge  
17                                   Hillcrest Village  
18                        Fairview, Henry Farm, Oriole  
19                                     Bayview Village  
20                            Silver Hills, York Mills  
21                             Newtonbrook, Willowdale  
22                                    Willowdale South  
23                                     York Mills West  
24                                     Willowdale West  
25                                           Parkwoods  
26                                     Don Mills North  
27                    Flemingdon Park, Don Mills South  
28     Bathurst Manor, Downsview North, Wilson Heights  

Therefore, the final **shape** is 103 rows by 3 columns.

## Part 2: Adding Location Data

In this part, I will add latitute and longitude data from a provided csv file. This will allow for future analysis of the region. 

In [15]:
import csv

csv_url = "http://cocl.us/Geospatial_data"
coordinates = pd.read_csv(csv_url)

print(coordinates.shape)
display(coordinates.head())

(103, 3)


Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [16]:
# Merge the two dataframes

df_full = pd.merge(df2, coordinates, on = "Postal Code")
print(df_full.shape)
display(df_full.head())

(103, 5)


Postal Code      Borough                            Neighborhood   Latitude  \
0         M1B  Scarborough                          Rouge, Malvern  43.806686   
1         M1C  Scarborough  Highland Creek, Rouge Hill, Port Union  43.784535   
2         M1E  Scarborough       Guildwood, Morningside, West Hill  43.763573   
3         M1G  Scarborough                                  Woburn  43.770992   
4         M1H  Scarborough                               Cedarbrae  43.773136   

   Longitude  
0 -79.194353  
1 -79.160497  
2 -79.188711  
3 -79.216917  
4 -79.239476

## Part 3: Explore and Cluster

Now we can explore the dataset and generate maps to visualize the neighborhoods and how they cluster together. 

Let's explore **Central and Downtown Toronto.**

In [17]:

c_toronto = df_full[df_full['Borough'].str.contains("Central Toronto")].reset_index()
d_toronto = df_full[df_full['Borough'].str.contains("Downtown Toronto")].reset_index()
cd_toronto = pd.concat([c_toronto, d_toronto])
display(cd_toronto)

index Postal Code           Borough  \
0      44         M4N   Central Toronto   
1      45         M4P   Central Toronto   
2      46         M4R   Central Toronto   
3      47         M4S   Central Toronto   
4      48         M4T   Central Toronto   
5      49         M4V   Central Toronto   
6      63         M5N   Central Toronto   
7      64         M5P   Central Toronto   
8      65         M5R   Central Toronto   
0      50         M4W  Downtown Toronto   
1      51         M4X  Downtown Toronto   
2      52         M4Y  Downtown Toronto   
3      53         M5A  Downtown Toronto   
4      54         M5B  Downtown Toronto   
5      55         M5C  Downtown Toronto   
6      56         M5E  Downtown Toronto   
7      57         M5G  Downtown Toronto   
8      58         M5H  Downtown Toronto   
9      59         M5J  Downtown Toronto   
10     60         M5K  Downtown Toronto   
11     61         M5L  Downtown Toronto   
12     66         M5S  Downtown Toronto   
13     67         M5T  Downtown Toronto   
14     68         M5V  Downtown Toronto   
15     69         M5W  Downtown Toronto   
16     70         M5X  Downtown Toronto   
17     75         M6G  Downtown Toronto   

                                         Neighborhood   Latitude  Longitude  
0                                       Lawrence Park  43.728020 -79.388790  
1                                    Davisville North  43.712751 -79.390197  
2                                  North Toronto West  43.715383 -79.405678  
3                                          Davisville  43.704324 -79.388790  
4                         Moore Park, Summerhill East  43.689574 -79.383160  
5   Deer Park, Forest Hill SE, Rathnelly, South Hi...  43.686412 -79.400049  
6                                            Roselawn  43.711695 -79.416936  
7                 Forest Hill North, Forest Hill West  43.696948 -79.411307  
8                 The Annex, North Midtown, Yorkville  43.672710 -79.405678  
0                                            Rosedale  43.679563 -79.377529  
1                         Cabbagetown, St. James Town  43.667967 -79.367675  
2                                Church and Wellesley  43.665860 -79.383160  
3                           Harbourfront, Regent Park  43.654260 -79.360636  
4                            Ryerson, Garden District  43.657162 -79.378937  
5                                      St. James Town  43.651494 -79.375418  
6                                         Berczy Park  43.644771 -79.373306  
7                                  Central Bay Street  43.657952 -79.387383  
8                            Adelaide, King, Richmond  43.650571 -79.384568  
9   Harbourfront East, Toronto Islands, Union Station  43.640816 -79.381752  
10           Design Exchange, Toronto Dominion Centre  43.647177 -79.381576  
11                     Commerce Court, Victoria Hotel  43.648198 -79.379817  
12                     Harbord, University of Toronto  43.662696 -79.400049  
13          Chinatown, Grange Park, Kensington Market  43.653206 -79.400049  
14  CN Tower, Bathurst Quay, Island airport, Harbo...  43.628947 -79.394420  
15                    Stn A PO Boxes 25 The Esplanade  43.646435 -79.374846  
16             First Canadian Place, Underground city  43.648429 -79.382280  
17                                           Christie  43.669542 -79.422564

In [18]:
# Now we can drop the first two columns

drop_col = ['index', 'Postal Code']
cd_toronto_data = cd_toronto.drop(drop_col, 1)
display(cd_toronto_data)

Borough                                       Neighborhood  \
0    Central Toronto                                      Lawrence Park   
1    Central Toronto                                   Davisville North   
2    Central Toronto                                 North Toronto West   
3    Central Toronto                                         Davisville   
4    Central Toronto                        Moore Park, Summerhill East   
5    Central Toronto  Deer Park, Forest Hill SE, Rathnelly, South Hi...   
6    Central Toronto                                           Roselawn   
7    Central Toronto                Forest Hill North, Forest Hill West   
8    Central Toronto                The Annex, North Midtown, Yorkville   
0   Downtown Toronto                                           Rosedale   
1   Downtown Toronto                        Cabbagetown, St. James Town   
2   Downtown Toronto                               Church and Wellesley   
3   Downtown Toronto                          Harbourfront, Regent Park   
4   Downtown Toronto                           Ryerson, Garden District   
5   Downtown Toronto                                     St. James Town   
6   Downtown Toronto                                        Berczy Park   
7   Downtown Toronto                                 Central Bay Street   
8   Downtown Toronto                           Adelaide, King, Richmond   
9   Downtown Toronto  Harbourfront East, Toronto Islands, Union Station   
10  Downtown Toronto           Design Exchange, Toronto Dominion Centre   
11  Downtown Toronto                     Commerce Court, Victoria Hotel   
12  Downtown Toronto                     Harbord, University of Toronto   
13  Downtown Toronto          Chinatown, Grange Park, Kensington Market   
14  Downtown Toronto  CN Tower, Bathurst Quay, Island airport, Harbo...   
15  Downtown Toronto                    Stn A PO Boxes 25 The Esplanade   
16  Downtown Toronto             First Canadian Place, Underground city   
17  Downtown Toronto                                           Christie   

     Latitude  Longitude  
0   43.728020 -79.388790  
1   43.712751 -79.390197  
2   43.715383 -79.405678  
3   43.704324 -79.388790  
4   43.689574 -79.383160  
5   43.686412 -79.400049  
6   43.711695 -79.416936  
7   43.696948 -79.411307  
8   43.672710 -79.405678  
0   43.679563 -79.377529  
1   43.667967 -79.367675  
2   43.665860 -79.383160  
3   43.654260 -79.360636  
4   43.657162 -79.378937  
5   43.651494 -79.375418  
6   43.644771 -79.373306  
7   43.657952 -79.387383  
8   43.650571 -79.384568  
9   43.640816 -79.381752  
10  43.647177 -79.381576  
11  43.648198 -79.379817  
12  43.662696 -79.400049  
13  43.653206 -79.400049  
14  43.628947 -79.394420  
15  43.646435 -79.374846  
16  43.648429 -79.382280  
17  43.669542 -79.422564

Now we can pull in more location data from the geocoding library. This will allow us to create a map of these boroughs in relation to Toronto as a whole. 

In [19]:
import folium

In [30]:
# Toronto latitude and longitude
latitute = 43.7170226
longitute = -79.4197830350134


# Generating a map of Toronto
map_toronto = folium.Map(location=[43.7170226, -79.4197830350134], zoom_start = 10)

# Adding markers for the neighborhoods
for latitude, longitude, borough, neighborhood in zip(cd_toronto_data['Latitude'], cd_toronto_data['Longitude'], cd_toronto_data['Borough'], cd_toronto_data['Neighborhood']):
    label = "{}, {}".format(neighborhood, borough)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [latitude, longitude],
        radius = 5, 
        popup = label,
        color = "purple",
        fill = True,
        fill_color = "#8f52e4", 
        fill_opacity = 0.6,
        parse_html = False).add_to(map_toronto)
    
map_toronto

## Foursquare API Integration

Now we can pull in data from Foursquare. 

Let's look at the coffee shops in the region.

In [35]:
CLIENT_ID = "Z5GRDXCYA1MPMN1QIZ2VN23EJLT40LFMPGL5ETQLCW2PW3JG"
CLIENT_SECRET = "PDVT5KZVEUERMIBZCQZA5ZBIQTQIN22RRMZJZTT1N1BMPJ1H"
VERSION = "20190608"
venue = "4bf58dd8d48988d196941735"

In [38]:
def getNearbyVenues(names, latitudes, longitudes, radius = 500):
    
    venues_list = []
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = "https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}".format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius)
            
        # make the GET request
        results = requests.get(url).json()["response"]["groups"][0]["items"]
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v["venue"]["name"], 
            v["venue"]["location"]["lat"], 
            v["venue"]["location"]["lng"],  
            v["venue"]['categories'][0]["name"]) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ["Neighborhood", 
                  "Neighborhood Latitude", 
                  "Neighborhood Longitude", 
                  "Venue", 
                  "Venue Latitude", 
                  "Venue Longitude", 
                  "Venue Category"]
    
    return(nearby_venues)

In [39]:
all_venues = getNearbyVenues(names = cd_toronto_data["Neighborhood"],
                            latitudes = cd_toronto_data['Latitude'],
                            longitudes = cd_toronto_data['Longitude'])

Lawrence Park
Davisville North
North Toronto West
Davisville
Moore Park, Summerhill East
Deer Park, Forest Hill SE, Rathnelly, South Hill, Summerhill West
Roselawn
Forest Hill North, Forest Hill West
The Annex, North Midtown, Yorkville
Rosedale
Cabbagetown, St. James Town
Church and Wellesley
Harbourfront, Regent Park
Ryerson, Garden District
St. James Town
Berczy Park
Central Bay Street
Adelaide, King, Richmond
Harbourfront East, Toronto Islands, Union Station
Design Exchange, Toronto Dominion Centre
Commerce Court, Victoria Hotel
Harbord, University of Toronto
Chinatown, Grange Park, Kensington Market
CN Tower, Bathurst Quay, Island airport, Harbourfront West, King and Spadina, Railway Lands, South Niagara
Stn A PO Boxes 25 The Esplanade
First Canadian Place, Underground city
Christie


In [40]:
print(all_venues.shape)
display(all_venues)

(585, 7)


Neighborhood  Neighborhood Latitude  \
0                             Lawrence Park              43.728020   
1                             Lawrence Park              43.728020   
2                             Lawrence Park              43.728020   
3                          Davisville North              43.712751   
4                          Davisville North              43.712751   
5                          Davisville North              43.712751   
6                          Davisville North              43.712751   
7                          Davisville North              43.712751   
8                          Davisville North              43.712751   
9                          Davisville North              43.712751   
10                         Davisville North              43.712751   
11                       North Toronto West              43.715383   
12                       North Toronto West              43.715383   
13                       North Toronto West              43.715383   
14                       North Toronto West              43.715383   
15                       North Toronto West              43.715383   
16                       North Toronto West              43.715383   
17                       North Toronto West              43.715383   
18                       North Toronto West              43.715383   
19                       North Toronto West              43.715383   
20                       North Toronto West              43.715383   
21                       North Toronto West              43.715383   
22                       North Toronto West              43.715383   
23                       North Toronto West              43.715383   
24                       North Toronto West              43.715383   
25                       North Toronto West              43.715383   
26                       North Toronto West              43.715383   
27                               Davisville              43.704324   
28                               Davisville              43.704324   
29                               Davisville              43.704324   
..                                      ...                    ...   
555  First Canadian Place, Underground city              43.648429   
556  First Canadian Place, Underground city              43.648429   
557  First Canadian Place, Underground city              43.648429   
558  First Canadian Place, Underground city              43.648429   
559  First Canadian Place, Underground city              43.648429   
560  First Canadian Place, Underground city              43.648429   
561  First Canadian Place, Underground city              43.648429   
562  First Canadian Place, Underground city              43.648429   
563  First Canadian Place, Underground city              43.648429   
564  First Canadian Place, Underground city              43.648429   
565  First Canadian Place, Underground city              43.648429   
566  First Canadian Place, Underground city              43.648429   
567  First Canadian Place, Underground city              43.648429   
568  First Canadian Place, Underground city              43.648429   
569  First Canadian Place, Underground city              43.648429   
570                                Christie              43.669542   
571                                Christie              43.669542   
572                                Christie              43.669542   
573                                Christie              43.669542   
574                                Christie              43.669542   
575                                Christie              43.669542   
576                                Christie              43.669542   
577                                Christie              43.669542   
578                                Christie              43.669542   
579                                Christie              43.669542   
580                                Christ

In [48]:
cd_toronto_coffee = all_venues[all_venues['Venue Category'] == "Coffee Shop"]

display(cd_toronto_coffee)

Neighborhood  Neighborhood Latitude  \
14                                  North Toronto West              43.715383   
20                                  North Toronto West              43.715383   
33                                          Davisville              43.704324   
47                                          Davisville              43.704324   
63   Deer Park, Forest Hill SE, Rathnelly, South Hi...              43.686412   
65   Deer Park, Forest Hill SE, Rathnelly, South Hi...              43.686412   
86                 The Annex, North Midtown, Yorkville              43.672710   
98                 The Annex, North Midtown, Yorkville              43.672710   
99                 The Annex, North Midtown, Yorkville              43.672710   
130                        Cabbagetown, St. James Town              43.667967   
133                        Cabbagetown, St. James Town              43.667967   
158                               Church and Wellesley              43.665860   
167                          Harbourfront, Regent Park              43.654260   
178                          Harbourfront, Regent Park              43.654260   
179                          Harbourfront, Regent Park              43.654260   
180                          Harbourfront, Regent Park              43.654260   
181                          Harbourfront, Regent Park              43.654260   
195                          Harbourfront, Regent Park              43.654260   
213                           Ryerson, Garden District              43.657162   
230                                     St. James Town              43.651494   
236                                     St. James Town              43.651494   
279                                        Berczy Park              43.644771   
286                                 Central Bay Street              43.657952   
287                                 Central Bay Street              43.657952   
288                                 Central Bay Street              43.657952   
304                                 Central Bay Street              43.657952   
307                                 Central Bay Street              43.657952   
309                                 Central Bay Street              43.657952   
314                                 Central Bay Street              43.657952   
327                           Adelaide, King, Richmond              43.650571   
381           Design Exchange, Toronto Dominion Centre              43.647177   
386           Design Exchange, Toronto Dominion Centre              43.647177   
395           Design Exchange, Toronto Dominion Centre              43.647177   
404           Design Exchange, Toronto Dominion Centre              43.647177   
411                     Commerce Court, Victoria Hotel              43.648198   
426                     Commerce Court, Victoria Hotel              43.648198   
428                     Commerce Court, Victoria Hotel              43.648198   
458                     Harbord, University of Toronto              43.662696   
469          Chinatown, Grange Park, Kensington Market              43.653206   
528                    Stn A PO Boxes 25 The Esplanade              43.646435   
544             First Canadian Place, Underground city              43.648429   
551             First Canadian Place, Underground city              43.648429   
561             First Canadian Place, Underground city              43.648429   
572                                           Christie              43.669542   

     Neighborhood Longitude                                Venue  \
14               -79.405678                            Starbucks   
20               -79.405678                          Tim Hortons   
33               -79.388790                            Starbucks   
47               -79.388790                           Second Cup   
63               -79.400049                            S

Lets also group these coffee shops by the neighborhoods we are looking at.

In [52]:
coffee_counts = cd_toronto_coffee.groupby("Neighborhood").count()
drop_cols = ['Neighborhood Longitude', 'Venue', 'Venue Latitude', 'Venue Longitude', 'Venue Category']
coffee_counts = coffee_counts.drop(drop_cols, 1)
coffee_counts.rename(columns = {"Neighborhood":"Neighborhood",
                              "Neighborhood Latitude":"Number of Coffee Shops"},
                    inplace = True)
display(coffee_counts)

Number of Coffee Shops
Neighborhood                                                              
Adelaide, King, Richmond                                                 1
Berczy Park                                                              1
Cabbagetown, St. James Town                                              2
Central Bay Street                                                       7
Chinatown, Grange Park, Kensington Market                                1
Christie                                                                 1
Church and Wellesley                                                     1
Commerce Court, Victoria Hotel                                           3
Davisville                                                               2
Deer Park, Forest Hill SE, Rathnelly, South Hil...                       2
Design Exchange, Toronto Dominion Centre                                 4
First Canadian Place, Underground city                                   3
Harbord, University of Toronto                                           1
Harbourfront, Regent Park                                                6
North Toronto West                                                       2
Ryerson, Garden District                                                 1
St. James Town                                                           2
Stn A PO Boxes 25 The Esplanade                                          1
The Annex, North Midtown, Yorkville                                      3